File: back/beatonma-django/github/events.py
CREATE_EVENT = "CreateEvent"
WIKI_EVENT = "GollumEvent"  # Wiki edits
ISSUES_EVENT = "IssuesEvent"
PULL_REQUEST_EVENT = "PullRequestEvent"
PUSH_EVENT = "PushEvent"
RELEASE_EVENT = "ReleaseEvent"


def all_events():
    return [
        CREATE_EVENT,
        PUSH_EVENT,
        ISSUES_EVENT,
        RELEASE_EVENT,
        WIKI_EVENT,
        PULL_REQUEST_EVENT,
    ]

File: front/src/apps/main/js/apps/github-events/types/events.ts
import { Repository } from "./common";
import { EventPayload } from "./payload";

export const Events = {
    Create: "CreateEvent",
    Push: "PushEvent",
    PullRequest: "PullRequestEvent",
    Wiki: "GollumEvent",
    Issue: "IssuesEvent",
    Release: "ReleaseEvent",
};

export interface Event {
    type: string;
    created_at: string;
}

export type PrivateEvent = Event;

export type PublicEvent = Event & {
    id: string;
    repository: Repository;
    payload: EventPayload;
};

export const isPublicEvent = (event: Event): event is PublicEvent => {
    if (event === null) return false;
    return (event as PublicEvent).id !== undefined;
};

export const isPrivateEvent = (event: Event): event is PrivateEvent => {
    if (event === null) return false;
    return !isPublicEvent(event);
};
